sktime/lagged_model_validator.py

```diff
@@ -0,0 +1,276 @@
+        mlags = np.asarray(mlags, dtype='i')
+        if np.any(mlags > maxmlag):
+            mlags = mlags[np.where(mlags <= maxmlag)]
+        if np.any(mlags < 0):
+            mlags = mlags[np.where(mlags >= 0)]
+
+        self.mlags = mlags
+
+    def _create_model(self) -> LaggedModelValidation:
+        return LaggedModelValidation()
+
+    def fit(self, data):
+        # set lag times
+        try:
+            input_lagtime = self.test_estimator.lagtime
+        except AttributeError:
+            try:
+                input_lagtime = self.test_model.lagtime
+            except AttributeError:
+                raise RuntimeError('Neither provided model nor estimator provides the "lagtime" attribute. Cannot proceed.')
+        self._set_mlags(data, input_lagtime)
+        lags = self.mlags * input_lagtime
+
+        predictions = []
+        predictions_conf = []
+        estimates = []
+        estimates_conf = []
+        estimated_models = []
+
+        # do we have zero lag? this must be treated separately
+        include0 = self.mlags[0] == 0
+        if include0:
+            lags = lags[1:]
+            estimated_models.append(None)
+            estimates.append(self._observable_dummy_mlag0())
+            predictions.append(self._observable_dummy_mlag0())
+            if self.has_errors:
+                estimates_conf.append(self._observable_dummy_mlag0(conf=True))
+                predictions.append(self._observable_dummy_mlag0(conf=True))
+
+        # estimate models at multiple of input lag time.
+        for lag in lags:
+            self.test_estimator.lagtime = lag
+            self.test_estimator.fit(data)
+            estimated_models.append(self.test_estimator.fetch_model().copy())
+
+        for mlag in self.mlags:
+            if mlag == 0: continue
+            # make a prediction using the test model
+            predictions.append(self._compute_observables(self.test_model, mlag=mlag))
+            # compute prediction errors if we can
+            if self.has_errors:
+                l, r = self._compute_observables_conf(self.test_model, mlag=mlag)
+                predictions_conf.append((l, r))
+
+        for model in estimated_models:
+            if model is None: continue
+            # evaluate the estimate at lagtime*mlag
+            estimates.append(self._compute_observables(model))
+            if self.has_errors and self.err_est:
+                l, r = self._compute_observables_conf(model)
+                estimates_conf.append((l, r))
+
+        # build arrays
+        estimates = np.array(estimates)
+        predictions = np.array(predictions)
+        if self.has_errors:
+            predictions_conf = np.array(predictions_conf)
+        else:
+            predictions_conf = np.array((None, None))
+        if self.has_errors and self.err_est:
+            estimates_conf = np.array(estimates_conf)
+        else:
+            estimates_conf = np.array((None, None))
+
+        m = self._model
```

Comment:
do we want to create a new empty model instance here?

Reply:
is this a rhetoric questions? :)

Reply:
maybe?

Reply:
we will get a new model instance, because it is a fit invocation.